src/VecSim/algorithms/brute_force/brute_force.cpp

```diff
@@ -43,24 +42,27 @@ size_t BruteForceIndex::estimateInitialSize(const BFParams *params) {
     est += (params->metric == VecSimMetric_L2 ? sizeof(L2Space) : sizeof(InnerProductSpace)) +
            sizeof(size_t);
     // Parameters related part.
-    est += params->initialCapacity * sizeof(decltype(idToVectorBlockMemberMapping)::value_type) +
-           sizeof(size_t);
+
+    if (params->initialCapacity) {
+        est += params->initialCapacity * sizeof(decltype(idToLabelMapping)::value_type) +
+               sizeof(size_t);
+    }
 
     return est;
 }
 
 size_t BruteForceIndex::estimateElementMemory(const BFParams *params) {
-    return params->dim * sizeof(float) + sizeof(labelType);
+    return params->dim * sizeof(float) + sizeof(idType);
 }
 
 void BruteForceIndex::updateVector(idType id, const void *vector_data) {
+
     // Get the vector block
-    VectorBlockMember *vectorBlockMember = this->idToVectorBlockMemberMapping[id];
-    VectorBlock *vectorBlock = vectorBlockMember->block;
-    size_t index = vectorBlockMember->index;
+    VectorBlock *vectorBlock = getVectorVectorBlock(id);
+    size_t index = getVectorRelativeIndex(id);
+
     // Update vector data in the block.
-    float *destinaion = vectorBlock->getVector(index);
-    memcpy(destinaion, vector_data, this->dim);
```

Comment:
This was a bug! Good catch! Also, I would point to this fix in the PR description...

Reply:
and add in the tests that the new vector blob was updated correctly

Reply:
tested in:
- test_delete_swap_block 
- brute_force_vector_update_test